### PR TITLE
[12.0][IMP] account_asset_management  - add days_calc

### PR DESCRIPTION
**Pull request by Saran440**
Add option to calculate depreciation table by days

### CODE REVIEW

**Comment by kittiu** [comment]
@Saran440 why close?

**Comment by pedrobaeza** [comment]
My mistake. Someone points this as a superseeded branch.

**Review by luc-demeyer — CHANGES_REQUESTED** [review]
can you rename this PR to "[12.0][IMP] account_asset_management - add days_calc"

**Comment by luc-demeyer** [comment]
@Saran440 
Can you have a look to the requested changes ?

**Comment by kittiu** [comment]
@Saran440 can you look at the conflict? May be you need to rebase?

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Functional and technical review. Thanks for the improvement! I backported to 11.0 here https://github.com/OCA/account-financial-tools/pull/902

File: account_asset_management/models/account_asset.py [hunk 1]
```diff
@@ -793,15 +800,25 @@ def _compute_line_dates(self, table, start_date, stop_date):
...
+            self, table, line_dates, depreciation_start_date,
+            depreciation_stop_date):
         digits = self.env['decimal.precision'].precision_get('Account')
         fy_residual_amount = self.depreciation_base
         i_max = len(table) - 1
         asset_sign = self.depreciation_base >= 0 and 1 or -1
+        day_amount = 0.0
+        if self.days_calc:
+            days = (depreciation_stop_date - depreciation_start_date).days + 1
+            day_amount = round(self.depreciation_base / days, digits)
```
Comment (JordiBForgeFlow) [inline account_asset_management/models/account_asset.py]: This rounding should not be applied. I have checked with this page and the numbers of daily depreciation do match between Microsoft and Odoo when this rounding is removed.

In Odoo, after removing the rounding.
image

In MS:
image

See it in my backport to 11.0 https://github.com/OCA/account-financial-tools/pull/902/commits/16de4642480e9c17a361f4d21765830e0179c0b3

**Comment by luc-demeyer** [comment]
@Saran440 
Can you fix the rounding issue so that we can proceed with  merging this PR ?

**Comment by JordiBForgeFlow** [comment]
@luc-demeyer @Saran440 I propose that we agree first on https://github.com/OCA/account-financial-tools/pull/902 and then we align the changes with this PR

**Comment by luc-demeyer** [comment]
@jbeficent 
Ok for me to first finish the work #902 and forward port that work later to 12.0.

**Comment by Saran440** [comment]
@luc-demeyer @jbeficent  Okay, I will fix the rounding issue. Thank you for suggest me

**Comment by Saran440** [comment]
@jbeficent Fixed. Ready to test.

**Comment by JordiBForgeFlow** [comment]
@Saran440 can you also include the other changes from #902, like the use_leap_year, so that both branches are in sync?

**Review by JoanSForgeFlow — APPROVED** [review]

File: account_asset_management/models/account_asset.py [hunk 2]
```diff
@@ -162,6 +162,11 @@ class AccountAsset(models.Model):
              # "  * Ending Date: Choose the time between 2 depreciations "
              # "and the date the depreciations won't go beyond."
     )
+    days_calc = fields.Boolean(
+        string='Calculate by days',
+        default=False,
```
Comment (nikul-serpentcs) [inline account_asset_management/models/account_asset.py:167]: @Saran440 No need to define `False`, Boolean always Default False 
https://www.odoo.com/documentation/10.0/reference/orm.html#odoo.fields.Field

File: account_asset_management/models/account_asset_profile.py [hunk 3]
```diff
@@ -90,6 +90,10 @@ class AccountAssetProfile(models.Model):
              "number of depreciation lines.\n"
              "  * Number of Years: Specify the number of years "
              "for the depreciation.\n")
+    days_calc = fields.Boolean(
+        string='Calculate by days',
+        default=False,
```
Comment (nikul-serpentcs) [inline account_asset_management/models/account_asset_profile.py:95]: same here

**Review by nikul-serpentcs — APPROVED** [review]
minor change, 
Otherwise code LGTM :+1:

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: technical and functional review.

**Comment by Saran440** [comment]
@jbeficent Thank you for your help. Should I rebase all PRs to 1 PR?

**Comment by JordiBForgeFlow** [comment]
Not necessary. I will merge now

**Comment by pedrobaeza** [comment]
Does this require migration scripts?

**Comment by kittiu** [comment]
@pedrobaeza thanks for your note. Please guide us.
@Saran440 would you check?

**Comment by JordiBForgeFlow** [comment]
No, does not require

**Pull request merged**